### PR TITLE
Fix 2 important bugs with Create Track, improve GzipBase64 names

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

File: map/src/context/TracksManager.js [hunk 1]
```diff
@@ -80,18 +80,25 @@ function openVisibleTracks(localTracks) {
 
 function saveLocalTrack(tracks, ctx) {
     let currentTrackIndex = tracks.findIndex(t => t.name === ctx.selectedGpxFile.name);
-    let track;
-    if (currentTrackIndex !== -1) {
-        track = ctx.selectedGpxFile;
-    } else {
-        track = tracks[tracks.length - 1];
+
+    if(currentTrackIndex === -1) {
```
Comment (alisa911) [inline map/src/context/TracksManager.js:84]: formatting

File: map/src/context/TracksManager.js [hunk 2]
```diff
@@ -80,18 +80,25 @@ function openVisibleTracks(localTracks) {
...
+        tracks.push(ctx.selectedGpxFile);
+        currentTrackIndex = tracks.findIndex(t => t.name === ctx.selectedGpxFile.name);
     }
+
+    if (currentTrackIndex === -1) {
```
Comment (alisa911) [inline map/src/context/TracksManager.js:89]: why this code

File: map/src/map/layers/LocalClientTrackLayer.js [hunk 3]
```diff
@@ -434,9 +441,9 @@ export default function LocalClientTrackLayer() {
         layers.addLayer(marker);
     }
 
-    function createLocalTrack(file, points) {
+    function createLocalTrack(file, points = [], wpts = []) {
         TracksManager.prepareTrack(file);
-        file.tracks = [{points: points}];
+        file.tracks = [{points: points, wpts: wpts}];
```
Comment (alisa911) [inline map/src/map/layers/LocalClientTrackLayer.js:446]: formatting

File: map/src/map/layers/LocalClientTrackLayer.js [hunk 4]
```diff
@@ -186,7 +189,11 @@ export default function LocalClientTrackLayer() {
 
     function saveLocal() {
         if (ctx.localTracks.length > 0) {
+            // localTracks exist: do update/append
```
Comment (alisa911) [inline map/src/map/layers/LocalClientTrackLayer.js:192]: delete comment

File: map/src/map/layers/LocalClientTrackLayer.js [hunk 5]
```diff
@@ -186,7 +189,11 @@ export default function LocalClientTrackLayer() {
...
             TracksManager.saveTracks(ctx.localTracks, ctx);
+        } else {
+            // localTracks empty: add gpx as 1st track (points and/or wpts are included)
```
Comment (alisa911) [inline map/src/map/layers/LocalClientTrackLayer.js:195]: delete comment

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2078

File: map/src/map/layers/LocalClientTrackLayer.js (hunk 4 above)
Comment (RZR-UA) [reply map/src/map/layers/LocalClientTrackLayer.js:192]: Comment describes ambiguous code: we must understand, why we calling createLocalTrack() in some case and saveTracks() in another.

There was a strange code which lead to the hidden bug, so this comments are really useful to remember the reasons of bug and also to undertand that saveTracks() don't work correctly if there is no tracks was saved before.

Finally, my opinion is to keep this kind of comments as useful for me and other developers.

File: map/src/map/layers/LocalClientTrackLayer.js (hunk 5 above)
Comment (RZR-UA) [reply map/src/map/layers/LocalClientTrackLayer.js:195]: Comment describes ambiguous code: we must understand, why we calling createLocalTrack() in some case and saveTracks() in another.

There was a strange code which lead to the hidden bug, so this comments are really useful to remember the reasons of bug and also to undertand that saveTracks() don't work correctly if there is no tracks was saved before.

Finally, my opinion is to keep this kind of comments as useful for me and other developers.

File: map/src/map/layers/LocalClientTrackLayer.js (hunk 3 above)
Comment (RZR-UA) [reply map/src/map/layers/LocalClientTrackLayer.js:446]: please post examples of correct formatting for this line

Comment (alisa911) [reply map/src/map/layers/LocalClientTrackLayer.js]: file.tracks = [
    {
        points: points, 
        wpts: wpts
    }
];

Comment (alisa911) [reply map/src/map/layers/LocalClientTrackLayer.js]: Screenshot 2023-06-30 at 08 43 29

**Pull request merged**